Open Judge System/Workers/OJS.Workers.ExecutionStrategies/DotNetCoreProjectTestsExecutionStrategy.cs

```diff
@@ -1,106 +1,149 @@
+            using System.Reflection;
+            using NUnit.Common;
+            using NUnitLite;
+            public class Program
+            {
+                public static void Main(string[] args)
+                {
+                    var writter = new ExtendedTextWrapper(Console.Out);
+                    new AutoRun(typeof(Program).GetTypeInfo().Assembly).Execute(args, writter, Console.In);
+                }
+            }";
+
+        private readonly string nunitLiteConsoleAppCsProjTemplate = $@"
+            <Project Sdk=""Microsoft.NET.Sdk"">
+                <PropertyGroup>
+                    <OutputType>Exe</OutputType>
+                    <TargetFramework>netcoreapp2.0</TargetFramework>
+                </PropertyGroup>
+                <ItemGroup>
+                    <PackageReference Include=""NUnitLite"" Version=""3.8.1"" />
+                    <PackageReference Include=""Microsoft.EntityFrameworkCore.InMemory"" Version=""2.0.0"" />
+                </ItemGroup>
+                <ItemGroup>
+                    {ProjectReferencesPlaceholder}
+                </ItemGroup>
+            </Project>";
+
+        private readonly string projectReferenceTemplate =
+            $@"<ProjectReference Include=""{ProjectPathPlaceholder}"" />";
+
+        public DotNetCoreProjectTestsExecutionStrategy(Func<CompilerType, string> getCompilerPathFunc)
+            : base(getCompilerPathFunc)
+        {
+        }
+
+        private string NUnitLiteConsoleAppDirectory =>
+            Path.Combine(this.WorkingDirectory, NUnitLiteConsoleAppFolderName);
+
+        private string UserProjectDirectory =>
+            Path.Combine(this.WorkingDirectory, UserSubmissionFolderName);
+
+        public override ExecutionResult Execute(ExecutionContext executionContext)
+        {
+            DirectoryHelpers.CreateDirecory(this.NUnitLiteConsoleAppDirectory);
+            DirectoryHelpers.CreateDirecory(this.UserProjectDirectory);
+
+            var result = new ExecutionResult();
+
+            var userSubmission = executionContext.FileContent;
+
+            this.ExtractFilesInWorkingDirectory(userSubmission, this.UserProjectDirectory);
+            this.ExtractTestNames(executionContext.Tests);
+
+            this.WriteTestFiles(executionContext.Tests, this.NUnitLiteConsoleAppDirectory);
+            this.WriteSetupFixture(this.NUnitLiteConsoleAppDirectory);
+
+            var userCsProjPaths = FileHelpers.FindAllFilesMatchingPattern(
+                this.UserProjectDirectory, CsProjFileSearchPattern);
+
+            var nunitLiteConsoleAppCsProjPath = this.CreateNunitLiteConsoleApp(
+                NUnitLiteConsoleAppProgramTemplate,
+                this.nunitLiteConsoleAppCsProjTemplate,
+                this.NUnitLiteConsoleAppDirectory,
+                userCsProjPaths);
+
+            var compilerPath = this.GetCompilerPathFunc(executionContext.CompilerType);
+
+            var consoleAppCompilerResult = this.Compile(
```

Comment:
`compilerResult` would suffice. It doesn't matter that the main project we're compiling here is our own console app, if the compilation fails it will be due to the user's project failing to compile since it's going to be compiled as well